File: Baekjoon/구현/10810.py
# https://www.acmicpc.net/problem/10810

import sys
input = sys.stdin.readline

n, m = map(int, input().split())

arr = [0] * n

for _ in range(m):
    i, j, k = map(int, input().split())
    arr[i - 1:j] = [k] * (j - i + 1)
    
print(*arr)

File: Baekjoon/구현/10810.cpp
// https://www.acmicpc.net/problem/10810

#include <iostream>
#include <vector>
using namespace std;

int main() {
    int n, m;
    cin >> n >> m;

    vector<int> baskets(n, 0);
    for (int i = 0; i < m; i++) {
        int a, b, c;
        cin >> a >> b >> c;

        for (int idx = a - 1; idx < b; idx++) {
            baskets[idx] = c;
        }
    }

    for (int num : baskets) {
        cout << num << " ";
    }
    cout << endl;

    return 0;
}